clients/geth/specular/rollup/services/sequencer/sequencer.go

```diff
@@ -62,37 +62,138 @@ func New(eth services.Backend, proofBackend proof.Backend, cfg *services.Config,
 	return s, nil
 }
 
+// Get tx index in batch
+func getTxIndexInBatch(slice []*types.Transaction, elem *types.Transaction) int {
+	for i := len(slice) - 1; i >= 0; i-- {
+		if slice[i].Hash() == elem.Hash() {
+			return i
+		}
+	}
+	return -1
+}
+
+// Filters tx in batch if already exists or on chain else appends
+func (s *Sequencer) modifyTxnsInBatch(batchTxs []*types.Transaction, tx *types.Transaction) []*types.Transaction {
+	// Check if tx in batch
```

Comment:
why you remove tx from batchTxs if it exists in batchTxs or the chain? shouldn't you just discard tx?